pkg/maps/egressmap/policy.go

```diff
@@ -33,41 +37,66 @@ type EgressPolicyVal4 struct {
 	GatewayIP types.IPv4 `align:"gateway_ip"`
 }
 
-// egressPolicyMap is the internal representation of an egress policy map.
-type egressPolicyMap struct {
-	*ebpf.Map
+var Cell = cell.Module(
+	"egressmaps",
+	"Egressmaps provide access to the egress gateway datapath maps",
+	cell.Config(DefaultPolicyConfig),
+	cell.Provide(func(daemonConfig *option.DaemonConfig, cfg PolicyConfig) (*PolicyMap, error) {
+		if !daemonConfig.EnableIPv4EgressGateway {
+			return nil, nil
+		}
+
+		return CreatePolicyMap(cfg)
+	}),
+)
+
+type PolicyConfig struct {
+	// EgressGatewayPolicyMapMax is the maximum number of entries
+	// allowed in the BPF egress gateway policy map.
+	EgressGatewayPolicyMapMax int
 }
 
-// initEgressPolicyMap initializes the egress policy map.
-func initEgressPolicyMap(policyMapName string, maxPolicyEntries int, create bool) error {
-	var m *ebpf.Map
+var DefaultPolicyConfig = PolicyConfig{
+	EgressGatewayPolicyMapMax: 1 << 14,
+}
 
-	if create {
-		m = ebpf.NewMap(&ebpf.MapSpec{
-			Name:       policyMapName,
-			Type:       ebpf.LPMTrie,
-			KeySize:    uint32(unsafe.Sizeof(EgressPolicyKey4{})),
-			ValueSize:  uint32(unsafe.Sizeof(EgressPolicyVal4{})),
-			MaxEntries: uint32(maxPolicyEntries),
-			Pinning:    ebpf.PinByName,
-		})
+func (def PolicyConfig) Flags(flags *pflag.FlagSet) {
```

Comment:
Can you point me to some Hive code or documentation which indicates how this Flag is initialized (how this is invoked). It seems strange to bury a user-facing flag in this file.